app/Controllers/Evaluation.php

```diff
@@ -5,59 +5,329 @@
 use App\Controllers\BaseController;
 
 use App\Models\SectionModel;
+use App\Models\EvalAnswersModel;
+use App\Models\QuestionchoiceModel;
+use App\Models\EvalSheetModel;
+use App\Models\EvalquestionModel;
+use App\Models\UserModel;
+use App\Models\EvaluationModel;
 
 class Evaluation extends BaseController
 {
-  public function index() {
+  public function _remap($method, $param1=null)
+  {
     if (!$this->session->has('logged_user')) {
       return redirect()->to(base_url('login'));
     } elseif (!$_SESSION['logged_user']['emailVerified']) {
       return redirect()->to(base_url('verifyAccount'));
     }
-
-    return redirect()->to(base_url('category'));
+    switch ($method)
+    {
+      case 'set_status':
+        if ($_SESSION['logged_user']['role'] === '2')
+          return redirect()->to(base_url('dashboard'));
+        else
+          return $this->$method();
+      case 'getUnfinished':
+        if ($_SESSION['logged_user']['role'] === '2')
+          return redirect()->to(base_url('dashboard'));
+        else
+          return $this->$method($param1);
+        break;
+      case 'evaluate':
+      case 'index':
+        if ($_SESSION['logged_user']['role'] === '1')
+          return redirect()->to(base_url('dashboard'));
+        else
+          return $this->evaluate($param1);
+      }
   }
 
-  public function set_status() {
+  public function set_status()
+  {
     return view('evaluation/setStatus');
   }
 
-  public function set_category($evalSheetId = null) {
-    $css = ['custom/modalAddition.css', 'custom/alert.css'];
-    $js = ['custom/alert.js'];
-
+  public function evaluate($eval_sheet_id = null)
+  {
     $data = [];
+    
+    if ($this->request->getMethod() === 'post') {
```

Comment:
- Checking if ang status of that **eval_sheet is already "completed"** then dapat di na siya maka himo ug changes for that subject na evaluation
- Default status is **"Open" or "open for evaluation"**, not **null**